### PR TITLE
logger + logsink

### PR DESCRIPTION
**Pull request by sebv**
- All logging call redirected to npmlog.
- logsink listens to all npm log call and redirect it to one winston instance.

requires https://github.com/appium/logger/pull/1

ping @jlipps for review.

### CODE REVIEW

File: lib/logsink.js [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    date = new Date(date.valueOf() + date.getTimezoneOffset() * 60000);
+  }
+  return date.toFormat("YYYY-MM-DD HH24:MI:SS:LL");
+};
+
+// Strip the color marking within messages.
+// We need to patch the transports, because the stripColor functionality in
+// Winston is wrongly implemented at the logger level, and we want to avoid
+// having to create 2 loggers.
+function applyStripColorPatch(transport) {
```
Comment (jlipps) [inline lib/logsink.js]: space before `(`

File: lib/logsink.js [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    let code = /\u001b\[(\d+(;\d+)*)?m/g;
+    msg = ('' + msg).replace(code, '');
+    _log(level, msg, meta, callback);
+  };
+}
+
+let _createConsoleTransport = function (args, logLvl) {
+  let transport = new (winston.transports.Console)({
+    name: "console"
+    , timestamp: args.logTimestamp ? timestamp : undefined
```
Comment (jlipps) [inline lib/logsink.js]: I think we should standardize on comma-last, now that trailing commas are supported

**Comment by jlipps** [comment]
:+1: for the approach, made a few style comments

File: lib/logsink.js (hunk 1 above)
Comment (sebv) [inline lib/logsink.js]: We don't usually put space after function name. https://github.com/appium/appium/blob/master/lib/server/helpers.js#L290

File: lib/logsink.js (hunk 2 above)
Comment (sebv) [inline lib/logsink.js]: yes comma last is better.

File: lib/logsink.js (hunk 1 above)
Comment (jlipps) [inline lib/logsink.js]: i thought we did. we have been doing it in all our new projects as well

**Pull request merged**